### DISCUSSION TITLE
Cleanup old podcasts/maintaining only most recent episode?

### DISCUSSION THREAD

**Discussion by bhcompy** [akhilrex/podgrab#216] [Ideas]
Are there any plans to add an option to only maintain the most recent episode(s)?  Or are there any tools you may be aware of to manage this?  I'm still a novice at searching for these kind of things.  

Use case: I subscribe to a number of daily news/politics podcasts and anything past this week is something I'll never really listen to, so if there was a way to clean this up that would be helpful(plus it helps with disk space management).  For example, ChannelsDVR has something like this for television recording where only the most recent episodes are kept.

**Comment by akhilrex** [comment]
Let me think about it.

On Thu, 16 Jun 2022 at 02:56, bhcompy ***@***.***> wrote:

> Are there any plans to add an option to only maintain the most recent
> episode(s)? Or are there any tools you may be aware of to manage this? I'm
> still a novice at searching for these kind of things.
>
> Use case: I subscribe to a number of daily news/politics podcasts and
> anything past this week is something I'll never really listen to, so if
> there was a way to clean this up that would be helpful(plus it helps with
> disk space management). For example, ChannelsDVR has something like this
> for television recording where only the most recent episodes are kept.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/akhilrex/podgrab/discussions/216>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEMVPQTWLBD7U5OIE2GHBTVPJDANANCNFSM5Y4WNTTQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Thanks and Regards,
Akhil Gupta

**Comment by da-wi** [comment]
If anyone is interested, I created a simple cronjob that does this job for me. In order for this to be working, I assume that you have checked the "Append episode number to episode file name" option, then this script will delete all podcast episodes with the lowest numbers and keep the `x` highest. This was for me somewhat safer then only go for the date modified - which would be another option and would just be a minor change in the "ls" command. 

Create the script:
`sudo nano /opt/podgrab/clean_directories.sh`
```
#!/bin/bash
declare -a podcasts=("Name of Podcast 1" "Name of Podcast 2")

for i in "${podcasts[@]}"
do
   ls -dr "/your/podgrab/directory/$i/"*  | grep .mp3 | tail -n +7 | sed 's/\ /\\ /g' | xargs rm -- 2> /dev/null
done
```
For explanation, the number in 
`tail -n +7`
actually gives the number of episodes to keep (i.e. 7-1=6 -> keep 6 episodes). 

Then call crontab:
`sudo crontab -e`

and add the line:
`0 0  * * * /opt/podgrab/clean_directories.sh` 

This makes the cleaning script run at midnight each day.
